src/main/src/test/java/org/geoserver/catalog/ResourcePoolTest.java

```diff
@@ -775,4 +777,20 @@ public void testEmptySort() throws IOException, IllegalAccessException {
             fc = (SimpleFeatureCollection) delegate;
         }
     }
+
+    @Test
+    public void testDefaultGeometry() throws IOException {
+        FeatureTypeInfo featureType =
+                getCatalog().getResourceByName("cdf", "Nulls", FeatureTypeInfo.class);
```

Comment:
@tbarsballe Where is the SimpleFeatureType for "cdf" defined?

Reply:
The data is here: https://github.com/geoserver/geoserver/blob/master/src/main/src/test/java/org/geoserver/data/test/Nulls.properties

The actual featureType gets cached in the resourceStore here: https://github.com/geoserver/geoserver/blob/master/src/main/src/main/java/org/geoserver/catalog/ResourcePool.java#L960-L961

Line 960 calculates the value from the store, while 961 calls a utility method to recalculate the feature type based on the schema declared in GeoServer, but doesn't handle defaultGeometry (that is where this fix is)

Reply:
Thanks for the clarification and the fix!  Looks good to me, but I don't know anywhere else that this might come up or impact.

Reply:
Yep, I'm a bit unsure of that myself. Thanks for taking a look at the PR.